DynamicAnalyzer/views/android/dynamic_analyzer.py

```diff
@@ -44,33 +55,41 @@ def dynamic_analysis(request):
                    ' Please run an android instance and refresh'
                    ' this page. If this error persists,'
                    ' set ANALYZER_IDENTIFIER in MobSF/settings.py')
-            return print_n_send_error_response(request, msg)
+            return print_n_send_error_response(request, msg, api)
         proxy_ip = get_proxy_ip(identifier)
-        context = {'apks': apks,
+        context = {'apps': scan_apps,
                    'identifier': identifier,
                    'proxy_ip': proxy_ip,
                    'proxy_port': settings.PROXY_PORT,
                    'title': 'MobSF Dynamic Analysis',
                    'version': settings.MOBSF_VER}
+        if api:
+            return context
         template = 'dynamic_analysis/dynamic_analysis.html'
         return render(request, template, context)
     except Exception as exp:
         logger.exception('Dynamic Analysis')
         return print_n_send_error_response(request,
-                                           exp)
+                                           exp,
+                                           api)
 
 
-def dynamic_analyzer(request):
+def dynamic_analyzer(request, api=False):
     """Android Dynamic Analyzer Environment."""
     logger.info('Creating Dynamic Analysis Environment')
     try:
-        bin_hash = request.GET['hash']
-        package = request.GET['package']
+        if api:
+            bin_hash = request.POST['hash']
+            package = request.POST['package']
```

Comment:
In various places the package is mandatory in addition to the hash, what is the reason for that?

Reply:
I have just APIfied the existing APIs. We use the package name at some place like for example https://github.com/MobSF/Mobile-Security-Framework-MobSF/pull/1560/files#diff-c705086180b959833bcd533f2b3395b5ebda6e04bf40bae4dc07ec99ab9a4c1aR144 to verify the installation of apk, start main activity of the apk etc.

We use logcat filtering by package name 
https://github.com/MobSF/Mobile-Security-Framework-MobSF/pull/1560/files#diff-c705086180b959833bcd533f2b3395b5ebda6e04bf40bae4dc07ec99ab9a4c1aR209

md5 hash we have is only known to MobSF, it does not have any relevance inside the dynamic analyzer. It only knows about package name.

Reply:
Yeah but the Dynamic analyzer could query for all the relevant info from the StaticAnalyzer.. isn't it better then making the user to make the query?

Reply:
Make sense. I can update that.